connect/transforms/src/test/java/org/apache/kafka/connect/transforms/HoistFieldTest.java

```diff
@@ -58,4 +60,17 @@ public void withSchema() {
         assertEquals(42, ((Struct) transformedRecord.key()).get("magic"));
     }
 
+    @Test
+    public void testSchemalessMapIsMutable() {
+        xform.configure(Collections.singletonMap("field", "magic"));
+
+        final SinkRecord record = new SinkRecord("test", 0, null, 420, null, null, 0);
+        final SinkRecord transformedRecord = xform.apply(record);
+
+        assertNull(transformedRecord.keySchema());
+        @SuppressWarnings("unchecked")
+        Map<String, Object> key = (Map<String, Object>) transformedRecord.key();
+        assertDoesNotThrow(() -> key.put("k", "v"));
```

Comment:
If we want to ensure that the map is actually mutable, wouldn't it make sense to also check that that the new key/value pair is present by reading it back?

```suggestion
        Map<String, Object> actualKey = (Map<String, Object>) transformedRecord.key();
        actualKey.put("k", "v");
        Map<String, Object> expectedKey = new HashMap<>();
        expectedKey.put("k", "v");
        expectedKey.put("magic", 420);
        assertEquals(expectedKey, actualKey);
```

Reply:
The idea behind this test is that the `put` operation would throw an `UnsupportedOperationException` if the map is immutable, so I don't think this explicit check is needed? Not a big deal though of course, I can always add the check if needed

Reply:
We get significantly stronger guarantees with very little additional complexity from this check, I think it's worth it.

Reply:
Fair enough, updated the tests 👍